### DISCUSSION TITLE
Adding an additional belong_to column to Noticed::Event

### DISCUSSION THREAD

**Discussion by scouillard** [excid3/noticed#420] [General]
Hi,

I would like to add a `tenant_id` column to the `noticed_event` table
`t.integer :tenant_id, type: :integer, as: "((params ->> 'tenant_id')::integer)", stored: true`

I have added the `belong_to` in the `event_extension`
```
  included do
    belongs_to :tenant
  end
```

However, this will trigger a validation error: `ActiveRecord::RecordInvalid: Validation failed: Tenant must exist`

I found a workaround, but I feel like I might be missing something?

Thank you very much.

**Comment by excid3** [comment]
You'll have to set the `tenant` association when you create the record. You can either set it on the model or override `with` to assign the tenant.

```ruby
notifier = MyNotifier.with(foo: :bar)
notifier.tenant = current_tenant
notifier.deliver(recipients)
```
